napari/_qt/_tests/test_plugin_widgets.py

```diff
@@ -191,3 +133,76 @@ def test_inject_viewer_proxy(make_napari_viewer):
     with patch('napari.utils.misc.ROOT_DIR', new='/some/other/package'):
         with pytest.warns(FutureWarning):
             wdg.fail()
+
+
+@pytest.mark.parametrize(
+    'widget_callable, param',
+    [
+        (QWidget_example, 'napari_viewer'),
+        (QWidget_string_annnot, 'test'),
+        (Container_example, 'test'),
+    ],
+)
+def test_get_widget_viewer_param(widget_callable, param):
+    """Test `_get_widget_viewer_param` returns correct parameter name."""
+    out = _get_widget_viewer_param(widget_callable, 'widget_name')
+    assert out == param
+
+
+def test_get_widget_viewer_param_error():
+    """Test incorrect subclass raises error in `_get_widget_viewer_param`."""
+    with pytest.raises(TypeError) as e:
+        _get_widget_viewer_param(ErrorWidget, 'widget_name')
+    assert "'widget_name' must be `QtWidgets.QWidget`" in str(e)
+
+
+def test_widget_hide_destroy(make_napari_viewer):
+    """Test that widget hide and destroy works."""
+    viewer = make_napari_viewer()
+    viewer.window.add_dock_widget(QWidget_example(viewer), name='test')
+    widget = viewer.window._dock_widgets['test']
+
+    # Check widget persists after hide
+    ww = widget.widget()
+    widget.title.hide_button.click()
+    assert ww
+    # Check that widget removed from `_dock_widgets` dict when closed
+    widget.destroyOnClose()
+    assert 'test' not in viewer.window._dock_widgets
+    assert ww.parent is None
```

Comment:
So in the original test this was adapted from, there was more widget clean up steps:

https://github.com/napari/napari/blob/27851ee92913e02b16d05b0aba19b126fa1e29e3/napari/_qt/_tests/test_plugin_widgets.py#L149-L155

are these needed?